pkg/metrics/builder/builder_test.go

```diff
@@ -0,0 +1,221 @@
+	}
+	sort.Strings(lpStrings)
+	return fmt.Sprintf("Desc{fqName: %q, help: %q, constLabels: {%s}, variableLabels: %v}",
+		opts.Name, opts.Desc, strings.Join(lpStrings, ","), []string{})
+}
+
+func getCollectorDescription(collector prometheus.Collector) string {
+	switch reflect.TypeOf(collector).Elem().Name() {
+	case "histogram":
+		return collector.(prometheus.Histogram).Desc().String()
+	case "counter":
+		return collector.(prometheus.Counter).Desc().String()
+	default:
+		return ""
+	}
+}
+
+func TestMetricBuilder(t *testing.T) {
+	cases := []struct {
+		name                  string
+		expectedCollectorType string
+		opts                  MetricOpts
+		labelKeyValues        map[string]string
+		value                 float64
+		expectedErrorMessage  string
+	}{
+		{
+			name:                  "Test histogram creation",
+			expectedCollectorType: "histogram",
+			opts: MetricOpts{
+				Labels:     []string{"test1", "test2"},
+				Desc:       "test histogram metrics",
+				Name:       "test_histogram",
+				Buckets:    []float64{10, 20, 30},
+				MetricType: Histogram,
+			},
+			labelKeyValues:       map[string]string{"test1": "test1", "test2": "test2"},
+			value:                0,
+			expectedErrorMessage: "",
+		},
+		{
+			name:                  "Test invalid labels",
+			expectedCollectorType: "histogram",
+			opts: MetricOpts{
+				Labels:     nil,
+				Desc:       "test metric",
+				Name:       "test",
+				MetricType: "Histogram",
+			},
+			labelKeyValues:       map[string]string{"test1": "test1", "test2": "test2"},
+			value:                0,
+			expectedErrorMessage: `labels cannot be empty`,
+		},
+		{
+			name:                  "Test invalid name",
+			expectedCollectorType: "histogram",
+			opts: MetricOpts{
+				Labels:     []string{"test1", "test2"},
+				Desc:       "test metric",
+				Name:       "",
+				MetricType: "Histogram",
+			},
+			labelKeyValues:       map[string]string{"test1": "test1", "test2": "test2"},
+			value:                0,
+			expectedErrorMessage: `name cannot be empty`,
+		},
+		{
+			name:                  "Test invalid metric type",
+			expectedCollectorType: "histogram",
+			opts: MetricOpts{
+				Labels:     []string{"test1", "test2"},
+				Desc:       "test metric",
+				Name:       "test",
+				MetricType: "",
+			},
+			labelKeyValues:       map[string]string{"test1": "test1", "test2": "test2"},
+			value:                0,
+			expectedErrorMessage: `metricType cannot be empty`,
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			_, err := NewMetricBuilder(tc.opts, tc.value, tc.labelKeyValues)
+			if tc.expectedErrorMessage == "" {
+				assert.NoError(t, err, "expected successful builder creation")
+			} else {
+				assert.Error(t, err, "expected builder creation failure but builder was created")
+				assert.EqualError(t, err, tc.expectedErrorMessage)
+			}
+		})
+	}
+}
+
+func TestMetricBuilderCollector(t *testing.T) {
```

Comment:
can we add a comment? 

TestMetricBuilderCollector tests the PromCollector function.